test/unit/org/apache/cassandra/cql3/RandomSchemaTest.java

```diff
@@ -0,0 +1,242 @@
+                    throw error;
+                }
+            }
+        });
+    }
+
+    private void maybeCreateUDTs(TableMetadata metadata)
+    {
+        Set<UserType> udts = CassandraGenerators.extractUDTs(metadata);
+        if (!udts.isEmpty())
+        {
+            Deque<UserType> pending = new ArrayDeque<>();
+            pending.addAll(udts);
+            Set<ByteBuffer> created = new HashSet<>();
+            while (!pending.isEmpty())
+            {
+                UserType next = pending.poll();
+                Set<UserType> subTypes = AbstractTypeGenerators.extractUDTs(next);
+                subTypes.remove(next); // it includes self
+                if (subTypes.isEmpty() || subTypes.stream().allMatch(t -> created.contains(t.name)))
+                {
+                    String cql = next.toCqlString(false, false);
+                    logger.warn("Creating UDT {}", cql);
+                    schemaChange(cql);
+                    created.add(next.name);
+                }
+                else
+                {
+                    logger.warn("Unable to create UDT {}; following sub-types still not created: {}", next.getCqlTypeName(), subTypes.stream().filter(t -> !created.contains(t.name)).collect(Collectors.toSet()));
+                    pending.add(next);
+                }
+            }
+        }
+    }
+
+    private static int primaryColumnCount(TableMetadata metadata)
+    {
+        return metadata.partitionKeyColumns().size() + metadata.clusteringColumns().size();
+    }
+
+    private String selectStmt(TableMetadata metadata)
+    {
+        StringBuilder sb = new StringBuilder();
+        sb.append("SELECT * FROM ").append(metadata).append(" WHERE ");
+        for (ColumnMetadata column : ImmutableList.<ColumnMetadata>builder().addAll(metadata.partitionKeyColumns()).addAll(metadata.clusteringColumns()).build())
+            sb.append(column.name.toCQLString()).append(" = ? AND ");
+        sb.setLength(sb.length() - " AND ".length());
+        return sb.toString();
+    }
+
+    private String insertStmt(TableMetadata metadata)
+    {
+        StringBuilder sb = new StringBuilder();
+        sb.append("INSERT INTO ").append(metadata.toString()).append(" (");
+        Iterator<ColumnMetadata> cols = metadata.allColumnsInSelectOrder();
+        while (cols.hasNext())
+            sb.append(cols.next().name.toCQLString()).append(", ");
+        sb.setLength(sb.length() - 2); // remove last ", "
+        sb.append(") VALUES (");
+        for (int i = 0; i < metadata.columns().size(); i++)
+        {
+            if (i > 0)
+                sb.append(", ");
+            sb.append('?');
+        }
+        sb.append(")");
+        return sb.toString();
+    }
+
+    private static Builder qt()
+    {
+        return new Builder();
+    }
+
+    public static class Builder
+    {
+        private long seed = System.currentTimeMillis();
```

Comment:
Can be `final`

Reply:
nope, `org.apache.cassandra.cql3.RandomSchemaTest.Builder#withFixedSeed` is dead code until we find a test failure; when an issue is found you call `withFixedSeed(failingSeed)` to rerun the test with the failing seed; this is why it can't be final